### PR TITLE
native simulator: minor additions

### PR DESCRIPTION
**Pull request by aescolar**
Four minor additions for the native simulator integration in Zephyr needed to support AMP targets.
(This is part of https://github.com/zephyrproject-rtos/zephyr/pull/63193, but can come on its own to ease review)

    native boards: AMP fix: localize the CONFIG_ symbols
    
    The Zephyr build leaves all kconfig options as absolute symbols
    in the image. This need to be localized, otherwise they will
    appear as duplicates with other Zephyr images.
    
-------

    native SOC: Add option to define how many MCUs a SOC has
    
    Add a new Kconfig option to define how many MCUs a SOC has
    
-----

    native SOC: Add option to select the primary MCU
    
    Add a new kconfig option to select which is the
    preffered embedded MCU.

-----

    native soc: Add option to pass extra images to native simulator build

    Add a new kconfig option to be able to pass extra images to the
    native simulator build.
    So one can, for ex., use one application build to produce one core image,
    and at the same time have it produce the final link with the native
    simulator runner and the other MCU images.

### CODE REVIEW

File: soc/posix/inf_clock/Kconfig
Comment (jori-nordic) [inline soc/posix/inf_clock/Kconfig:32]: for me a path sounds like something that should rather be in cmake, do you agree?

**Review by jori-nordic — DISMISSED** [review]

File: soc/posix/inf_clock/Kconfig
Comment (aescolar) [reply soc/posix/inf_clock/Kconfig:32]: discussed offline that this makes it easy for users to find and more evident that it is an option they can set it, and it is used similarly in quite a few other places.

File: soc/posix/inf_clock/Kconfig
Comment (Thalley) [inline soc/posix/inf_clock/Kconfig:34]: Redundant default: https://docs.zephyrproject.org/latest/build/kconfig/tips.html#redundant-defaults
```suggestion
```

File: soc/posix/inf_clock/Kconfig
Comment (Thalley) [inline soc/posix/inf_clock/Kconfig:22]: I guess this is based on some index, right? Maybe
```suggestion
config NATIVE_SIMULATOR_PRIMARY_MCU_INDEX
```
or
```suggestion
config NATIVE_SIMULATOR_PRIMARY_MCU_ID
```

**Review by Thalley — CHANGES_REQUESTED** [review]

File: soc/posix/inf_clock/Kconfig
Comment (aescolar) [reply soc/posix/inf_clock/Kconfig:22]: Do you feel strongly enough about this like to block the PR? These 2 options are not really for users but boards definitions.
(there is 20 commits on top of these ones.. so these rename are quite painful.)

File: soc/posix/inf_clock/Kconfig
Comment (aescolar) [reply soc/posix/inf_clock/Kconfig:34]: thanks will fix

File: soc/posix/inf_clock/Kconfig
Comment (aescolar) [reply soc/posix/inf_clock/Kconfig:37]: thanks will fix

File: soc/posix/inf_clock/Kconfig
Comment (Thalley) [reply soc/posix/inf_clock/Kconfig:22]: Let's get another set of eyes on this, and if someone else agrees that they are fine, we'll keep them as is :) 

@jori-nordic What do you think?

Comment (jori-nordic) [reply soc/posix/inf_clock/Kconfig:22]: I think they're fine. Maybe just in case you could make them promptless @aescolar ? That way we can rename later if really necessary.

Comment (aescolar) [reply soc/posix/inf_clock/Kconfig:22]: If I make them promptless it is not possible anymore to set them in defconfigs or project .confs
NATIVE_SIMULATOR_PRIMARY_MCU_N I expect users could want to change

Comment (Thalley) [reply soc/posix/inf_clock/Kconfig:22]: Alright, @aescolar please then just update the other comments and leave these as is

**Comment by aescolar** [comment]
This is the diff (ignoring rebase):
```diff
diff --git a/boards/posix/common/natsim_config.cmake b/boards/posix/common/natsim_config.cmake
index f370504509..576e4b9b63 100644
--- a/boards/posix/common/natsim_config.cmake
+++ b/boards/posix/common/natsim_config.cmake
@@ -19,7 +19,7 @@ set(nsi_config_content
   "NSI_LINK_OPTIONS:=$<JOIN:$<TARGET_PROPERTY:native_simulator,INTERFACE_LINK_OPTIONS>,\ >"
   "NSI_EXTRA_LIBS:=$<JOIN:$<TARGET_PROPERTY:native_simulator,RUNNER_LINK_LIBRARIES>,\ >"
   "NSI_PATH:=${NSI_DIR}/"
-  "NSI_N_CPUS:=${CONFIG_NATIVE_SIMULATOR_N_MCUS}"
+  "NSI_N_CPUS:=${CONFIG_NATIVE_SIMULATOR_NUMBER_MCUS}"
   "NSI_LOCALIZE_OPTIONS:=--localize-symbol=CONFIG_*"
 )
 
diff --git a/soc/posix/inf_clock/Kconfig b/soc/posix/inf_clock/Kconfig
index 686f9fe538..3f42f7b23d 100644
--- a/soc/posix/inf_clock/Kconfig
+++ b/soc/posix/inf_clock/Kconfig
@@ -11,7 +11,7 @@ config NATIVE_SIMULATOR_CPU_N
          This option is only applicable for targets which use the
          native simulator as their runner.
 
-config NATIVE_SIMULATOR_N_MCUS
+config NATIVE_SIMULATOR_NUMBER_MCUS
        int "Total number of MCUs this target has"
        range 1 16
        default 1
@@ -19,7 +19,7 @@ config NATIVE_SIMULATOR_N_MCUS
        help
          How many AMP MCUs does this target have in total.
 
-config NATIVE_SIMULATOR_PRIMARY_MCU_N
+config NATIVE_SIMULATOR_PRIMARY_MCU_INDEX
        int "Which CPU is the primary/preferred"
        default 0
        depends on NATIVE_LIBRARY
@@ -31,10 +31,9 @@ config NATIVE_SIMULATOR_PRIMARY_MCU_N
 
 config NATIVE_SIMULATOR_EXTRA_IMAGE_PATHS
        string "Other cores images to include"
-       default ""
        depends on NATIVE_LIBRARY
        help
-         This option can be used to provide the native simulator other MCUs/Cores images which have been
-         produced by either other Zephyr builds or different OS builds.
+         This option can be used to provide the native simulator with other MCUs/Cores images which have
+         been produced by either other Zephyr builds or different OS builds.
          So you can, for ex., use this application build produce one core image, and at the same time
          have it produce the final link with the native simulator runner and the other MCU images.
```

File: soc/posix/inf_clock/Kconfig
Comment (aescolar) [reply soc/posix/inf_clock/Kconfig]: I did the changes you requested... they were really painful..

**Review by Thalley — APPROVED** [review]

**Review by jori-nordic — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**
